aten/src/ATen/templates/TensorMethods.h

```diff
@@ -21,6 +22,30 @@
 
 namespace at {
 
+namespace detail {
+
+struct MultiDispatchTensorTypeSet : IterArgs<MultiDispatchTensorTypeSet> {
+  TensorTypeSet ts;
+  void operator()(const at::Tensor& x) {
+    ts = ts | x.type_set();
+  }
+  void operator()(TensorOptions x) {
+    ts = ts | x.type_set();
+  }
+  void operator()(at::ArrayRef<at::Tensor> xs) {
+    for (const auto& x : xs) {
```

Comment:
so is it official that dispatch looks into tensor lists beyond the first element?

Reply:
Yes, it has to. Suppose you have lazy tensor and you stack a some normal tensors with lazy tensor; you have to get to the lazy tensor version in that case.